### ISSUE TITLE
add speaker-page support for special characters in city name

### ISSUE THREAD

**Issue by bridgetkromhout** [devopsdays/devopsdays-web#858]
Right now, https://github.com/devopsdays/devopsdays-web/commit/38c67842b8196a3d3d036d641e6abd8dbbbb8d18 fixed speaker and talk pages to allow multi-word cities but still don't cover special characters. This needs to be fixed before Brasília tries to create their program (to be announced October 8th).

One approach - store the result of https://github.com/devopsdays/devopsdays-web/blob/master/make_new_event.sh#L11 so we don't have to pull it out of the event slug? (Or parse the event slug?)

**Comment by mattstratton** [comment]
This is basically reverted to using `friendly` again, right?

**Comment by bridgetkromhout** [comment]
I wouldn't put it that way. "friendly" is awfully ambiguous. I think we need "the city half of the slug" to be easy to grab. I _also_ think maybe we need to think through whether `urlize` is what we want to be using there after all. It would bring in dashes (and thus require some aliasing probably) but maybe that's better for seo anyhow?

**Comment by mattstratton** [comment]
Oh yeah, you need the "with spaces" version of the city.

It seems like it's best to pull it from the slug, since you'll know where to get it from (since it will always start at position 6). I kind of don't want to add even more metadata to the data files, since we then have to go back and retcon all the data files. And since with #133 we are gonna touch all that code then, I don't really want to do it now?

Multi-word cities are supposed to use dashes and not "smushed" together anyway for their path (i.e., it should be `salt-lake-city`, not `saltlakecity`), but I can see we DID NOT do them that way. I'd rather we try to move to the dashes because also, SEO.

**Comment by mattstratton** [comment]
Since you go ahead and generate it anyway, I'm thinking that it's okay to stick the value of `$cityabbr` from the script into a data file element called `city-slug` or something. And then in the speaker page code, you basically do a check for that element existing - if it does, it uses that, if not, it uses the value of `city`.

Looking back through code, I'm getting really confused. We have some super unnecessary code in the `future` and `past` partials that check for a value in the data called `displayname`, but we ended up not having to use it because we refactored out ever called `city` for talking to a path.
